rasa/core/policies/ted_policy.py

```diff
@@ -347,22 +355,32 @@ def __init__(
         self.only_e2e = TEXT in self.fake_features and INTENT not in self.fake_features
 
         self._label_data: Optional[RasaModelData] = None
-        self.data_example: Optional[Dict[Text, List[np.ndarray]]] = None
+        self.data_example: Optional[Dict[Text, Dict[Text, List[FeatureArray]]]] = None
 
         self.tmp_checkpoint_dir = None
         if self.config[CHECKPOINT_MODEL]:
             self.tmp_checkpoint_dir = Path(rasa.utils.io.create_temporary_directory())
 
+    @staticmethod
+    def model_class() -> Type[RasaModel]:
+        return TED
+
+    @classmethod
+    def _metadata_filename(cls) -> Optional[Text]:
+        return SAVE_MODEL_FILE_NAME
```

Comment:
It might be similar to one of my previous comments but what's the reason of returning a constant from a class method? Especially since this class method is private.

Reply:
Again, `_load_model_utilities` relies on it and `SAVE_MODEL_FILE_NAME` needs to be of the correct class.